### ISSUE TITLE
Enhance state schema

### ISSUE THREAD

**Issue by daanboer** [LXCat-project/LXCat#58]
### TLDR

These changes introduce three main benefits:
 1. They make it so that it is a lot more clear at first sight which type of state (singular, compound, unspecified) you are dealing with.
 2. The resulting state definitions are more concise.
 3. The arbitrary `e`, `v`, and `J` properties of unspecified states are eliminated.

### Explanation

The current state schema roughly distinguishes between three different types of states: singular, compound, and unspecified. To allow for the schema to express each of these types it is setup in a generic way. For example, electronic, vibrational, and rotational subcomponents are always presented as an array, whereas this is not always necessary. I will present three examples from the test set to illustrate this.

<details>
<summary> Singular: $\mathrm{N2}\left(\mathrm{X}^{1}\Sigma_\mathrm{g}^+\left(v=0\left(J=24\right)\right)\right)$
</summary>

```json
{
  "particle": "N2",
  "charge": 0,
  "type": "HomonuclearDiatom",
  "electronic": [
    {
      "e": "X",
      "Lambda": 0,
      "S": 0,
      "parity": "g",
      "reflection": "+",
      "vibrational": [
        {
          "v": 0,
          "rotational": [{ "J": 24 }]
        }
      ]
    }
  ]
}
```
</details>

<details><summary>Compound: $\mathrm{CO2}\left(\mathrm{X}^{1}\Sigma_\mathrm{g}^+\left(v=0,0,0|0,5,0|2,1,0|1,3,0|0,2,1|1,0,1\right)\right)$</summary>

```json
{
  "particle": "CO2",
  "charge": 0,
  "type": "LinearTriatomInversionCenter",
  "electronic": [
    {
      "e": "X",
      "Lambda": 0,
      "S": 0,
      "parity": "g",
      "reflection": "+",
      "vibrational": [
        { "v": [0, 0, 0] },
        { "v": [0, 5, 0] },
        { "v": [2, 1, 0] },
        { "v": [1, 3, 0] },
        { "v": [0, 2, 1] },
        { "v": [1, 0, 1] }
      ]
    }
  ]
}
```
</details>

<details><summary>Unspecified: $\mathrm{Ar}\left(*\right)$</summary>

```json
{
  "particle": "Ar",
  "charge": 0,
  "type": "AtomLS",
  "electronic": [
    {
      "e": "*"
    }
  ]
}
```
</details>

To clarify, I am talking about the `electronic`, `vibrational`, and `rotational` properties. Their values are always presented as an array, whereas this only really makes sense for compound states. For example, for singular states the straight brackets can be omitted such that the only entry in the array is directly assigned to the relevant property. This is demonstrated in the following adjusted example:

<details><summary>Singular: $\mathrm{N2}\left(\mathrm{X}^{1}\Sigma_\mathrm{g}^+\left(v=0\left(J=24\right)\right)\right)$</summary>

```json
{
  "particle": "N2",
  "charge": 0,
  "type": "HomonuclearDiatom",
  "electronic": {
    "e": "X",
    "Lambda": 0,
    "S": 0,
    "parity": "g",
    "reflection": "+",
    "vibrational": {
      "v": 0,
      "rotational": { "J": 24 }
    }
  }
}
```
</details>

For the compound example, the `electronic` array can be collapsed, but the `vibrational` property is still presented by an array as this state is a vibrational compound. One big question is whether we even want to allow multi-level compounds (that is, states that provide multiple children on multiple levels, e.g.  electronic and vibrational)? I currently see little advantage in allowing for their existence, and supporting them can imply more work for surrounding tools (e.g. how would these states be stored and linked in the database?). Although perhaps there are some edge cases that would require such states? 

<details><summary>Compound: $\mathrm{CO2}\left(\mathrm{X}^{1}\Sigma_\mathrm{g}^+\left(v=0,0,0|0,5,0|2,1,0|1,3,0|0,2,1|1,0,1\right)\right)$</summary>

```json
{
  "particle": "CO2",
  "charge": 0,
  "type": "LinearTriatomInversionCenter",
  "electronic": {
    "e": "X",
    "Lambda": 0,
    "S": 0,
    "parity": "g",
    "reflection": "+",
    "vibrational": [
      { "v": [0, 0, 0] },
      { "v": [0, 5, 0] },
      { "v": [2, 1, 0] },
      { "v": [1, 3, 0] },
      { "v": [0, 2, 1] },
      { "v": [1, 0, 1] }
    ]
  }
}
```
</details>

The unspecified states can be simplified even further. Currently, an unspecified component is an object with a single property that stores the string definition. The key of this property differs per level, e.g.: `e`; for electronic, `v`; for vibrational, and `J`; for rotational. This was originally done such that when the state definition is parsed, the `summary` and `latex` properties can be added to this object. However, these object are superfluous for unspecified components as they are always equal to the original string identifier. Therefore, the added indirection is unnecessary, and the string identifier can be directly assigned to the relevant property. This results in a much more concise state definition:

<details><summary>Unspecified: $\mathrm{Ar}\left(*\right)$</summary>

```json
{
  "particle": "Ar",
  "charge": 0,
  "type": "Unspecified",
  "electronic": "*"
}
```
</details>

where I use the `Unspecified` type as proposed in #57.

**Comment by daanboer** [comment]
Closed by #547
